### PR TITLE
Bookstore: Use Redux in React components

### PR DESCRIPTION
**Pull request by obibaadoma**
Wrap your <App> with the <Provider> component from react-redux
Pass your configured store into the <Provider>
Display your books, received from the slice, in a reusable component
Dispatch actions using useDispatch
Add a <Button> component, which includes:

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import './index.css';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <BrowserRouter>
+      <App />
+    </BrowserRouter>
+  </React.StrictMode>,
+);
```
Comment (Strangeal) [inline src/index.js]: - [ ] Great job so far. However, your project is missing some important parts of the project requirements. To fulfill the project requirements kindly consider wrapping your `App` with the `<Provider>` component from `react-redux` as mentioned in the project requirements and pass your `store` to it. See below for more details on project requirements:
Screenshot 2023-08-22 at 23 02 59

File: src/redux/categoriesSlice.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+import categoryReducer from './categories/categories';
+
+const store = configureStore({
+  reducer: {
+    books: booksReducer,
+    categories: categoryReducer,
+  },
+});
+
+export default store;
```
Comment (Strangeal) [inline src/redux/categoriesSlice.js]: - [`OPTIONAL` ] Great job creating your store. Everything looks good so far but for the sake of conversion I would like to suggest that you change the name of this file from `categoriesSlice` to `store`.

File: src/components/BookList.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      completed: true,
+      id: uuid4(),
+    },
+    {
+      title: 'ancestral sacrifice',
+      author: 'akosombo nyantakyi',
+      completed: true,
+      id: uuid4(),
+    },
+  ];
```
Comment (Strangeal) [inline src/components/BookList.js]: - [ ] Kindly remove this list of books you have in this file as you don't need it. You're expected to make use of `useSelector` from `react-redux` to get all books from your `booksSlice` and populate it here nicely as you have done.

File: src/components/BookForm.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      <h2>{title}</h2>
+      <form>
+        <input type="text" placeholder="Add title" className="input-text" name="title" />
+        <input type="text" placeholder="Add author" className="input-text" name="author" />
+        <button type="submit">ADD BOOK</button>
+      </form>
+    </section>
+  );
+};
+
```
Comment (Strangeal) [inline src/components/BookForm.js]: - [ ] Kindly make sure of `useState` to manage the state of your form as mentioned in the project requirement. 
- [ ] Make sure to `dispatch` the `ADD_BOOKS` reducer from your `booksSlice` **onClick** of a the **ADD BOOK** `button`.

File: src/components/ContentBook.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const ContentBook = ({ title, author }) => (
+  <li>
+    <h3>{title}</h3>
+    <p>{author}</p>
+    <button type="button">Delete</button>
+  </li>
+);
```
Comment (Strangeal) [inline src/components/ContentBook.js]: - [ ] Kindly make use of `useDispatch` to dispatch `REMOVE_BOOKS` reducer from your bookSlice to **remove** a book from the `books` array (with the *id* attribute)

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @obibaadoma, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text

## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Correct `GitHub` workflow ✅ 
- Meaningful **Github** commits ✅ 
- Correct `.github/workflows` for linters ✅ 
- Followed *best-practices* ✅ 
- Correct **Semantics** used ✅ 
- Well structured project ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (obibaadoma) [reply src/index.js]: thanks for the suggestion is fixed.

File: src/redux/categoriesSlice.js (hunk 2 above)
Comment (obibaadoma) [reply src/redux/categoriesSlice.js]: name changed to store

File: src/components/BookList.js (hunk 3 above)
Comment (obibaadoma) [reply src/components/BookList.js]: removed as suggested

File: src/components/ContentBook.js (hunk 5 above)
Comment (obibaadoma) [reply src/components/ContentBook.js]: used dispatch

File: src/redux/books/books.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// Constants
+const ADD_BOOK = 'bookstore/src/redux/books/addBook';
+const REMOVE_BOOK = 'bookstore/src/redux/books/removeBOOK';
+
+// Action creators
+/* const initialBook = [
+  { id: 0, title: 'Operating Systems', author: 'xxx' },
+  { id: 1, title: 'Computer Architecture', author: 'william Stallings' },
+  { id: 2, title: 'Software Engineering', author: 'Robert Mitchil' },
+]; */
```
Comment (Munsa1) [inline src/redux/books/books.js]: - [ ] Great work implementing the previously required changes, your app works fine. However, due to the refactor you have left commented code in your application. Please remove the commented code for a cleaner codebase and a more professional looking one.

```
components folder files to consider
- book.js
- InputBook.js
```

```
redux folder files to consider
- books.js
- configureStore.js
```

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @obibaadoma, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
Code Has Been Refactored As Required :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 6 above)
Comment (obibaadoma) [reply src/redux/books/books.js]: code removed. thanks

File: src/components/book.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+Book.defaultProps = {
+  books: [],
+};
+
+/* Book.propTypes = {
+  books: Number,
+}; */
+
```
Comment (thecodechaser) [inline src/components/book.js]: - [ ] As requested by the previous reviewer, Please remove the commented code from this file. It's not good practice to leave commented code in the codebase.

File: src/components/InputBook.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidv4 } from 'uuid';
+import { addBook } from '../redux/books/books';
+
+// const initialId = 0;
+
```
Comment (thecodechaser) [inline src/components/InputBook.js]: - [ ] As requested by the previous reviewer, Please remove the commented code from this file. It's not good practice to leave commented code in the codebase.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @obibaadoma👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- Good job with redux. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/InputBook.js (hunk 8 above)
Comment (obibaadoma) [reply src/components/InputBook.js]: done

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/book.js (hunk 7 above)
Comment (obibaadoma) [reply src/components/book.js]: removed comment

**Pull request merged**